### PR TITLE
IPC-Compliant Two-pin landpattern generator

### PR DESCRIPTION
**Pull request by m-hilgendorf**
Example:

Screenshot from 2021-07-22 14-40-38

This PR adds a set of generators for creating IPC compliant two-pin landpatterns given the manufacturing and fabrication tolerances.

### CODE REVIEW

File: stanza.proj [hunk 1]
```diff
@@ -2,11 +2,11 @@
 ; Land pattern and symbol utilities 
 ;====================================
 package ocdb/land-patterns defined-in "utils/land-patterns.stanza"
-package ocdb/land-patterns/ipc-helpers defined-in "utils/land-patterns/ipc-helpers.stanza"
-package ocdb/land-patterns/leads defined-in "utils.land-patterns/ipc-helpers.stanza"
+package ocdb/land-protrusions defined-in "utils/land-protrusions.stanza"
```
Comment (m-hilgendorf) [inline stanza.proj:5]: rename old file to `land-protrusions`

File: tests/test-land-patterns.stanza [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+pcb-component example-2:
+  pin a
+  pin c
+  reference-prefix = "C"
+  pcb-landpattern lp:
+    make-two-pin-chip-landpattern(
+      Toleranced(3.2, 0.2),
+      Toleranced(1.6, 0.2),
+      Toleranced(0.8, 0.3),
+      Toleranced(1.2, 0.1),
```
Comment (m-hilgendorf) [inline tests/test-land-patterns.stanza:60]: This example has a polarity marker and override for lead-width that is smaller than the component-width.

File: utils/design-vars.stanza [hunk 3]
```diff
@@ -18,11 +21,15 @@ public var DESIGN-QUANTITY = 100
 public var PREFERRED-MOUNTING = "smd" ; values in ["smd", "through-hole"]
 public var MIN-CAP-VOLTAGE = 10.0 ; Minimum voltage to allow for capacitors
 
-; For consolidation
-;public var PREFERRED-MANUFACTURERS = ["Yageo", "Panasonic Electronic Components", "Vishay Dale", "TE Connectivity Passive Product"]
+; Landpattern variables
 public var SOLDER-MASK-REGISTRATION: Double = 0.15
+public var PLACEMENT-TOLERANCE   = 0.0 ; "P" on page 13 IPC-7351B. FIXME goes in pcb-rules
```
Comment (m-hilgendorf) [inline utils/design-vars.stanza:26]: Add better defaults

File: utils/land-patterns.stanza [hunk 4]
```diff
@@ -18,6 +20,62 @@ defn ensure-even-positive! (value: Int, name: String):
   if value < 2 or value % 2 == 1:
     fatal("%_ only supports a positive even number of pins." % [name])
 
+; find the angle of a pad away from the origin
+defn pad-angle (pad-ref:Ref) -> Double:
```
Comment (m-hilgendorf) [inline utils/land-patterns.stanza:24]: remove, unnecessary

File: utils/land-patterns.stanza [hunk 5]
```diff
@@ -93,6 +151,21 @@ defn chamfered (anchor:Anchor, width:Double, height:Double, r:Double):
 ;==============================================================================
 ;=========================== Geometric Helpers ================================
 ;==============================================================================
+; ====== Land pattern utilities ===============================================
+; Find the extent dimensions of a landpattern given all its layers and pads.
+public defn extent-dims () -> Dims:
```
Comment (m-hilgendorf) [inline utils/land-patterns.stanza:156]: sanity check current usage and remove if we aren't using it anymore

File: utils/land-patterns.stanza [hunk 6]
```diff
@@ -153,8 +226,8 @@ public defn apply-soldermask (amount:Double, expand?:True|False):
       (s:ChamferedRectangle|DShape|Capsule):
```
Comment (m-hilgendorf) [inline utils/land-patterns.stanza:226]: big TODO here: replace with new expand-by-amount function that is not equivalent to `offset` or `expand` (preserve shape, but increase dimensions additively rather than multiplicatively)

File: utils/land-patterns.stanza [hunk 7]
```diff
@@ -558,10 +634,16 @@ public defn make-ipc-two-pin-landpattern (part-name:String, courtyard?:True|Fals
         val nr = Z
         val nc = X
         [Z, G, X, nr, nc]
-    make-two-pin-landpattern(Z, G, X, nr, nc, courtyard?, polarized?)
+    if v1?:
```
Comment (m-hilgendorf) [inline utils/land-patterns.stanza:637]: delete this check

File: utils/land-patterns.stanza [hunk 8]
```diff
@@ -558,10 +634,16 @@ public defn make-ipc-two-pin-landpattern (part-name:String, courtyard?:True|Fals
...
+      make-two-pin-landpattern(Z, G, X, nr, nc, courtyard?, polarized?)
+    else:
+      make-two-pin-landpattern(Z, G, X, nr, nc, courtyard?, polarized?)
     external-names = ([part-name])
     ref-label()
 
+public defn make-ipc-two-pin-landpattern (part-name:String, courtyard?:True|False, polarized?:True|False):
```
Comment (m-hilgendorf) [inline utils/land-patterns.stanza:644]: remove

File: utils/tolerance.stanza [hunk 9]
```diff
@@ -0,0 +1,77 @@
+#use-added-syntax(jitx)
+defpackage ocdb/tolerance:
+  import core
+  import collections
+  import math
+
+; A generic value with some tolerance. May be offcenter (upper tolerance != lower tolerance)
+public pcb-struct ocdb/tolerance/Toleranced:
```
Comment (m-hilgendorf) [inline utils/tolerance.stanza:8]: Is `Toleranced` the appropriate name?

**Review by RajeshJITX — DISMISSED** [review]

File: utils/tolerance.stanza [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+; Return the upper tolerance, as a percentage
+public defn tol+% (t:Toleranced) -> Double:
+  plus(t) / nominal(t)
+
+; Return the lower tolerance, as a percentage
+public defn tol-% (t:Toleranced) -> Double:
+  minus(t) / nominal(t)
+
+; Return the max value
+public defn max (t:Toleranced) -> Double:
```
[the remark this answers is not in the document]
Comment (CuppoJava) [reply utils/tolerance.stanza:43]: Agreed.

[the remark this answers is not in the document]
Comment (m-hilgendorf) [reply utils/tolerance.stanza:43]: Any suggestions on an alternative name?

[the remark this answers is not in the document]
Comment (CuppoJava) [reply utils/tolerance.stanza:43]: `max-value` is one suggestion.

File: utils/land-patterns.stanza [hunk 11]
```diff
@@ -911,22 +965,22 @@ public pcb-landpattern INDC1005X55 :
 
 public pcb-landpattern CAPT1005X55 :
   make-ipc-two-pin-landpattern("0402")
-  layer(Silkscreen("pol", Top)) = PlusSymbol(-1.7, 0.0)
+  draw-polarity-marker(IndexRef(Ref("p"), 1))
```
Comment (CuppoJava) [inline utils/land-patterns.stanza:968]: After the change to `draw-polarity-marker` to take `JITXObject`, you can call it like this:

```
draw-polarity-marker(self.p[1])
```

Comment (CuppoJava) [reply utils/land-patterns.stanza:968]: It seems that the `two-pin-landpattern` should directly generate the polarity marker itself instead of relying upon callers to do it. Especially since it already takes a `polarized?` argument.

File: utils/land-patterns.stanza [hunk 12]
```diff
@@ -1034,3 +1088,155 @@ public pcb-landpattern smd-testpoint-pkg (d:Double) :
...
+
+public defn make-two-pin-chip-landpattern (length:Toleranced,
+                                           width:Toleranced,
+                                           lead-length:Toleranced,
+                                           lead-width?:False|Toleranced, ; in case different from width
+                                           fabrication-tolerance:Double,
+                                           placement-tolerance:Double
+                                           density-level:DensityLevel,
+                                           polarized?:True|False):
+                                          ;  solder-mask-defined?:True|False): ;
```
Comment (CuppoJava) [inline utils/land-patterns.stanza:1100]: Remove commented-out code from public-facing OCDB.

File: utils/tolerance.stanza [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+public pcb-struct ocdb/tolerance/Toleranced:
+  tol+:Double,
+  tol-:Double,
+  typ: Double,
+
+defmethod print (o:OutputStream, t:Toleranced):
+  println(o, "Toleranced(%_ <= %_ <= %_)" % [min-value(t), typ(t), max-value(t)])
+
+; Create a tolerance value with the same tolerance in both directions
+public defn Toleranced (value:Double, tolerance:Double) -> Toleranced:
```
Comment (CuppoJava) [inline utils/tolerance.stanza:17]: Use the same convention for all the `Toleranced` constructors. 

Either they are called `Toleranced` to follow standard Stanza convention, or they have their own short names (e.g. `tol%`).

**Review by CuppoJava — CHANGES_REQUESTED** [review]
Requested some cleanups.

File: utils/land-patterns.stanza [hunk 14]
```diff
@@ -18,6 +20,54 @@ defn ensure-even-positive! (value: Int, name: String):
   if value < 2 or value % 2 == 1:
     fatal("%_ only supports a positive even number of pins." % [name])
 
+; ====== Silk screen helpers ===================================================
+public defn draw-polarity-marker (pad-ref:Ref, part-dims?:Dims|False) :
```
[the remark this answers is not in the document]
Comment (m-hilgendorf) [reply utils/land-patterns.stanza:24]: part-dims are the physical dimensions of the part (not any of the layers associated with the land pattern). We don't have a way of representing this generally, so I pass them in as an argument.

File: utils/land-patterns.stanza (hunk 11 above)
Comment (m-hilgendorf) [reply utils/land-patterns.stanza:968]: This code does not call the two pin landpattern generator (and doing so would break it, because callers of this generator don't use `a` and `c` as the pad names).

File: utils/land-patterns.stanza [hunk 15]
```diff
@@ -201,6 +252,9 @@ public pcb-pad smd-pad (s:Shape):
   name  = "SMD Pad"
   type  = SMD
   shape = s
+  ; TODO: add the soldermask manually here with offset so it's
```
[the remark this answers is not in the document]
Comment (m-hilgendorf) [reply utils/land-patterns.stanza]: Removed

File: utils/land-patterns.stanza (hunk 11 above)
Comment (m-hilgendorf) [reply utils/land-patterns.stanza]: Added the changes

File: utils/land-patterns.stanza (hunk 12 above)
Comment (m-hilgendorf) [reply utils/land-patterns.stanza]: Done

File: utils/tolerance.stanza (hunk 13 above)
Comment (m-hilgendorf) [reply utils/tolerance.stanza]: Done

File: utils/design-vars.stanza [hunk 16]
```diff
@@ -1,14 +1,17 @@
...
-;==============================================================================
-;========================= Global Checks Parameters ===========================
-;==============================================================================
-public var OPERATING-TEMPERATURE = [0.0 25.0] ; the [min, max] operating temperature of the design, in Celsius.
+public pcb-enum ocdb/design-vars/DensityLevel:
+  DensityLevelA
+  DensityLevelB
+  DensityLevelC
+
+public var OPERATING-TEMPERATURE = [0.0 25.0]
```
Comment (bpellenard) [inline utils/design-vars.stanza:14]: Should we mention the unit?

File: utils/land-patterns.stanza [hunk 17]
```diff
@@ -18,6 +20,55 @@ defn ensure-even-positive! (value: Int, name: String):
   if value < 2 or value % 2 == 1:
     fatal("%_ only supports a positive even number of pins." % [name])
 
+; ====== Silk screen helpers ===================================================
+; Create a polarity marker relative to a pad.
+; pda
```
Comment (bpellenard) [inline utils/land-patterns.stanza:25]: pda?

File: utils/design-vars.stanza (hunk 16 above)
Comment (m-hilgendorf) [reply utils/design-vars.stanza:14]: I think we assume C

File: utils/land-patterns.stanza (hunk 17 above)
Comment (m-hilgendorf) [reply utils/land-patterns.stanza]: Good catch, fixed.

File: utils/land-patterns.stanza [hunk 18]
```diff
@@ -18,6 +20,55 @@ defn ensure-even-positive! (value: Int, name: String):
...
+        x(pad-dims) * -0.5 + x(center(pad-loc)) - SOLDER-MASK-REGISTRATION
+        y(pad-dims) *  0.5 + y(center(pad-loc)) + R
+      )
+    
+    val [p1, p2] = [pad-pose(), part-pose()] 
+    draw-line $
+      if distance(p1) > distance(p2):
+        p1
+      else: 
+        p2
```
Comment (bpellenard) [inline utils/land-patterns.stanza:69]: max(distance, [p1 p2])?

Comment (m-hilgendorf) [reply utils/land-patterns.stanza:69]: Which usage of `max` is this? I'm not familiar

Comment (bpellenard) [reply utils/land-patterns.stanza:69]: or maybe `maximum`

File: utils/tolerance.stanza [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+  import math
+
+; A generic value with some tolerance. May be offcenter (upper tolerance != lower tolerance)
+public pcb-struct ocdb/tolerance/Toleranced:
+  typ: Double,
+  tol+:Double,
+  tol-:Double,
+
+defmethod print (o:OutputStream, t:Toleranced):
+  println(o, "Toleranced(%_ <= %_ <= %_)" % [min-value(t), typ(t), max-value(t)])
```
Comment (bpellenard) [inline utils/tolerance.stanza:14]: `print`

File: utils/tolerance.stanza [hunk 20]
```diff
@@ -0,0 +1,60 @@
+#use-added-syntax(jitx)
+defpackage ocdb/tolerance:
+  import core
+  import collections
+  import math
+
+; A generic value with some tolerance. May be offcenter (upper tolerance != lower tolerance)
+public pcb-struct ocdb/tolerance/Toleranced:
+  typ: Double,
```
Comment (bpellenard) [inline utils/tolerance.stanza:9]: could it be 0.0?

File: utils/tolerance.stanza [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
+public defn tol (typ:Double, tol+:Double, tol-:Double) -> Toleranced:
+  Toleranced(typ, tol+, tol-)
+
+; Create a tolerance with the same upper and lower tolerances
+public defn tol (typ:Double, tol+-:Double) -> Toleranced:
+  tol(typ, tol+-, tol+-)
+
+; Return the upper tolerance, as a percentage
+public defn tol+% (t:Toleranced) -> Double:
+  tol+(t) / typ(t)
```
Comment (bpellenard) [inline utils/tolerance.stanza:36]: careful if `typ(t)` equals `0.0`

File: utils/tolerance.stanza [hunk 22]
```diff
@@ -0,0 +1,60 @@
...
+public defn max-value (t:Toleranced) -> Double:
+  typ(t) + tol-(t)
+
+; Return the min value
+public defn min-value (t:Toleranced) -> Double:
+  typ(t) - tol-(t)
+
+; Return the center of the value range
+public defn center (t:Toleranced) -> Double:
+  0.5 * (min-value(t) + max-value(t))
```
Comment (bpellenard) [inline utils/tolerance.stanza:52]: careful with overflow: `min-value(t) + 0.5 * (max-value(t) - min-value(t))`

File: utils/tolerance.stanza [hunk 23]
```diff
@@ -0,0 +1,60 @@
...
+
+; Check if a value is within the tolerance's range.
+public defn in-range? (t:Toleranced, value:Double) -> True|False:
+  value >= min-value(t) and value <= max-value(t)
+ 
+ ; Return the difference between max and min of the toleranced values
```
Comment (bpellenard) [inline utils/tolerance.stanza:58]: Returns?

File: utils/tolerance.stanza [hunk 24]
```diff
@@ -0,0 +1,60 @@
+#use-added-syntax(jitx)
+defpackage ocdb/tolerance:
```
Comment (bpellenard) [inline utils/tolerance.stanza:2]: should we create unit tests for the packages in this repo?

File: utils/tolerance.stanza (hunk 20 above)
Comment (m-hilgendorf) [reply utils/tolerance.stanza:9]: It's very unlikely. But we also can't use `with: (ensure)` in pcb-structs.

File: utils/tolerance.stanza (hunk 22 above)
Comment (m-hilgendorf) [reply utils/tolerance.stanza:52]: Even with doubles?

File: utils/tolerance.stanza (hunk 21 above)
Comment (m-hilgendorf) [reply utils/tolerance.stanza:36]: I think if `typ == 0.0` then this is still valid, the result would be NaN

File: utils/tolerance.stanza (hunk 19 above)
Comment (m-hilgendorf) [reply utils/tolerance.stanza]: done

File: utils/tolerance.stanza (hunk 22 above)
Comment (m-hilgendorf) [reply utils/tolerance.stanza]: done`

File: utils/tolerance.stanza (hunk 23 above)
Comment (m-hilgendorf) [reply utils/tolerance.stanza:63]: done

File: utils/tolerance.stanza (hunk 24 above)
Comment (m-hilgendorf) [reply utils/tolerance.stanza:2]: Done - also good call, it found a few bugs

File: utils/tolerance.stanza [hunk 25]
```diff
@@ -0,0 +1,65 @@
...
+; Create a tolerance with the same upper and lower tolerances
+public defn tol (typ:Double, tol+-:Double) -> Toleranced:
+  tol(typ, tol+-, tol+-)
+
+; Returns the upper tolerance, as a percentage
+public defn tol+% (t:Toleranced) -> Double:
+  tol+(t) / typ(t)
+
+; Returns the lower tolerance, as a percentage
+public defn tol-% (t:Toleranced) -> Double:
```
Comment (bpellenard) [inline utils/tolerance.stanza:44]: add unit tests for this function

File: utils/tolerance.stanza (hunk 21 above)
Comment (bpellenard) [reply utils/tolerance.stanza:41]: ok, I would like to see this edge case covered in the unit tests

File: utils/tolerance.stanza [hunk 26]
```diff
@@ -0,0 +1,65 @@
...
+; Returns the center of the value range
+public defn center (t:Toleranced) -> Double:
+  min-value(t) + 0.5 * (max-value(t) - min-value(t))
+
+; Check if a value is within the tolerance's range.
+public defn in-range? (t:Toleranced, value:Double) -> True|False:
+  value >= min-value(t) and value <= max-value(t)
+ 
+ ; Returns the difference between max and min of the toleranced values
+public defn tolerance-range (t:Toleranced) -> Double:
```
Comment (bpellenard) [inline utils/tolerance.stanza:64]: add unit test

File: utils/tolerance.stanza [hunk 27]
```diff
@@ -0,0 +1,65 @@
...
+  import collections
+  import math
+
+; A generic value with some tolerance. May be offcenter (upper tolerance != lower tolerance)
+public pcb-struct ocdb/tolerance/Toleranced:
+  typ: Double,
+  tol+:Double,
+  tol-:Double,
+
+defmethod print (o:OutputStream, t:Toleranced):
```
Comment (bpellenard) [inline utils/tolerance.stanza:13]: add in unit test, not as `print` but as `to-string`, and call `#EXPECT` on the result

Comment (m-hilgendorf) [reply utils/tolerance.stanza:13]: done

File: utils/tolerance.stanza (hunk 26 above)
Comment (m-hilgendorf) [reply utils/tolerance.stanza:64]: done

File: utils/tolerance.stanza (hunk 25 above)
Comment (m-hilgendorf) [reply utils/tolerance.stanza:44]: done

**Review by bpellenard — APPROVED** [review]
It looks great @m-hilgendorf! Thank you very much for the updates.

File: utils/tolerance.stanza (hunk 21 above)
Comment (CuppoJava) [reply utils/tolerance.stanza:41]: Better to crash with `fatal` than to return NaN.

Comment (m-hilgendorf) [reply utils/tolerance.stanza:41]: It actually returns infinity in this case. Should we still crash?

File: utils/land-patterns.stanza [hunk 28]
```diff
@@ -911,22 +966,22 @@ public pcb-landpattern INDC1005X55 :
...
 public pcb-landpattern CAPT2012X125 :
   make-ipc-two-pin-landpattern("0805")
-  layer(Silkscreen("pol", Top)) = PlusSymbol(-2.4, 0.0)
+  draw-polarity-marker(self.p[1])
   ref-label()
 
 public pcb-landpattern CAPT3216X160 :
   make-ipc-two-pin-landpattern("1206")
-  layer(Silkscreen("pol", Top)) = PlusSymbol(-3.0, 0.0)
+  draw-polarity-marker(self.p[1])
```
Comment (CuppoJava) [inline utils/land-patterns.stanza:984]: See later comment.

File: utils/land-patterns.stanza [hunk 29]
```diff
@@ -1034,3 +1089,154 @@ public pcb-landpattern smd-testpoint-pkg (d:Double) :
   layer(Courtyard(Top)) = Circle(d / 2.0)
   layer(Silkscreen("f-silk", Top)) = Circle(d / 2.0)
   ref-label()
+
+public defn make-two-pin-chip-landpattern (length:Toleranced,
```
Comment (CuppoJava) [inline utils/land-patterns.stanza:1093]: Please add a comment describing what this function is supposed to do. Then we can judge whether or not the API is appropriate. 

For example: Is this generator responsible for drawing the polarity marker when it is necessary? 

Currently it seems the answer is: It depends. 
1) When modeling a non-polarized component, this generator will handle the polarity marker appropriately. (By doing nothing.)
2) When modeling a polarized component, with pins a and c, this generator will handle the polarity marker appropriately. (By drawing it.)
3) When modeling a polarized component, with pins p[0] and p[1], (e.g. CAPT3216X160) then it is the caller's responsibility to pass `false` for the `polarized?` argument, and then manually call `draw-polarity-marker` themselves.

Point 3 when written out, makes it seem like this would be a tricky API to use.

File: utils/land-patterns.stanza [hunk 30]
```diff
@@ -18,6 +20,58 @@ defn ensure-even-positive! (value: Int, name: String):
   if value < 2 or value % 2 == 1:
     fatal("%_ only supports a positive even number of pins." % [name])
 
+; ====== Silk screen helpers ===================================================
+; Create a polarity marker relative to a pad.
+; 
+; pad-1: the pad to mark for the polarity 
+; part-dims?: optional. If provided, the generator will use these to determine where
```
Comment (CuppoJava) [inline utils/land-patterns.stanza:27]: This utility seems to assume that the current landing pattern is centered at the origin. Is that true? 

Can we remove this assumption? If not, this assumption makes this utility less applicable as a general user-facing utility.

File: utils/land-patterns.stanza [hunk 31]
```diff
@@ -18,6 +20,58 @@ defn ensure-even-positive! (value: Int, name: String):
...
+;             to place the pad, and guarantee it is not underneath the part.
+public defn draw-polarity-marker (pad-1:JITXObject, part-dims?:Dims|False) :
```
Comment (CuppoJava) [inline utils/land-patterns.stanza:29]: Can we have a screen shot of what this function is supposed to generate?

File: utils/land-patterns.stanza [hunk 32]
```diff
@@ -18,6 +20,58 @@ defn ensure-even-positive! (value: Int, name: String):
...
+    defn draw-line (origin:Pose):
+      val line-width  = min-silk-width
+      val line-length = line-width * 3.0
+      val line = Line(line-width, [
+        Point(min-silk-width * 0.5, 0.0)
+        Point(line-width + min-silk-width * 0.5, 0.0)  
+      ])
+      layer(Silkscreen("pol", Top)) = origin * line
+
+    defn part-pose () :
```
Comment (CuppoJava) [inline utils/land-patterns.stanza:43]: I'm not sure what the `part-dims?` argument is controlling here, can you describe it?

**Review by CuppoJava — CHANGES_REQUESTED** [review]
Clarification needed on some APIs.

File: utils/land-patterns.stanza (hunk 28 above)
Comment (CuppoJava) [reply utils/land-patterns.stanza:984]: Thanks for point out that these are actually calling: make-ipc-two-pin-landpattern instead of the newer make-two-pin-chip-landpattern.

File: utils/tolerance.stanza (hunk 21 above)
Comment (CuppoJava) [reply utils/tolerance.stanza:42]: Yes. Crashing is good. I cannot imagine a scenario where a user will rely upon the +Inf for his code to work.

Comment (m-hilgendorf) [reply utils/tolerance.stanza:42]: Fixed, these functions now call `fatal`.

File: utils/land-patterns.stanza [hunk 33]
```diff
@@ -1178,7 +1180,7 @@ public defn make-two-pin-chip-landpattern (length:Toleranced,
   make-two-pin-chip-landpattern(
     length, width, lead-length, lead-width?, 
     FABRICATION-TOLERANCE, PLACEMENT-TOLERANCE, 
-    density-level, polarized?,
+    density-level, polarized?, polarized?
```
Comment (bpellenard) [inline utils/land-patterns.stanza:1183]: are you going to use, in the future, `make-two-pin-chip-landpattern` with different values for `anode-and-cathode?` and `polarized?`?

Comment (m-hilgendorf) [reply utils/land-patterns.stanza:1158]: Yes.

**Comment by m-hilgendorf** [comment]
Updated screen shot showing polarity marker shape and distance:

Screenshot from 2021-07-26 12-26-56

**Review by CuppoJava — APPROVED** [review]
Looks great. Thank you!
